### ISSUE TITLE
Kotlin test coverage not reported for Java production code with JaCoCo

### ISSUE THREAD

**Issue by realdadfish** [Kotlin/kotlinx-kover#530]
This is my configuration:

```
plugins {
    id("org.jetbrains.kotlinx.kover")
}

val configuration = extensions.create("glscoverage", GlsCoverageExtension::class.java)

afterEvaluate {
    kover {
        // not working
        useJacoco("0.8.11")
    }

    val expected = configuration.expectedCoverage.get()
    val threshold = configuration.coverageThreshold.get()
    val minBound = (expected - threshold).coerceAtLeast(0)
    val maxBound = (expected + threshold).coerceAtMost(100)

    koverReport {
        defaults {
            mergeWith(configuration.buildVariant.get())
        }

        filters {
            excludes {
                annotatedBy("*Generated*")
                classes(configuration.classFilter.get())
            }
        }

        verify {
            rule {
                minBound(minBound)
                maxBound(maxBound)
            }
        }

        androidReports(configuration.buildVariant.get()) {}
    }

    tasks.withType(Test::class.java).configureEach {
        finalizedBy(tasks.named("koverXmlReport"))
    }
}
```

Now I have separate Kotlin and Java source sets for my mixed-language module configured, i.e. `src/{main,test}/{kotlin,java}` and I have a Kotlin test that tests various Java source files. With the above configuration the coverage for these Java source files is reported as 0, even if I move the test from `src/test/kotlin` to `src/test/java`, however, if I comment out `useJacoco` above and let the IJ coverage processor do it's thing, the coverage of the classes in question is properly reported.

**Environment**
 - Kover Gradle Plugin version: 0.7.5
 - Gradle version: 8.5
 - Kotlin project type: Kotlin/Android
 - Coverage Toolset (if customized in build script): Kover, Jacoco (0.8.11)

**Comment by realdadfish** [comment]
Didn't made a difference, same result. The coverage in Jacoco is still missing:
<img width="1343" alt="Bildschirmfoto 2024-01-29 um 18 24 15" src="https://github.com/Kotlin/kotlinx-kover/assets/208234/5c78c4de-fa73-4438-9d07-69ebab07c898">
<img width="1688" alt="Bildschirmfoto 2024-01-29 um 18 25 56" src="https://github.com/Kotlin/kotlinx-kover/assets/208234/ae7498f4-c308-4a4a-a790-ef625ec0d7f2">

**Comment by shanshin** [comment]
Is this a multi-module project?

**Comment by realdadfish** [comment]
The component in question has just one module, the app module. The convention plugin is applied on the app module level.

**Comment by shanshin** [comment]
Could you please provide a reproducer project?

**Comment by realdadfish** [comment]
Here it is: https://github.com/realdadfish/kover-issue-530/tree/main

It seems to be triggered by Robolectric on the test classpath. If I have two regular Java/Kotlin unit tests, both execute fine and the coverage shows that both methods are covered. If I let each test run with the Robolectric runner, _neither_ test shows coverage on the class with JaCoCo, but _both_ tests show coverage with the built-in coverage tool

**Comment by realdadfish** [comment]
Many thanks!

**Comment by shanshin** [comment]
Fixed in `0.7.6`
